### ISSUE TITLE
Internal Server Error on NC 22.2.3

### ISSUE THREAD

**Issue by mhorst00** [nextcloud/cookbook#855]
**Description**
When attempting to open the Cookbook tab in the Nextcloud browser interface, I only get the 'Internal Server Error' page. The app worked just fine and then stopped working

**Reproduction**
1. Go to Nextcloud webinterface
2. Click on 'Cookbook'

**Expected behavior**
I would expect to see my saved recipes

**Actual behavior**
Error page is shown

**Logs**
<details>
<summary>Nextcloud backend logs</summary>

```json
{
  "reqId": "cEosCBwSBIPgNnKFyO88",
  "level": 3,
  "time": "2021-11-23T11:14:21+00:00",
  "remoteAddr": "fcdd:1::6",
  "user": "User",
  "app": "index",
  "method": "GET",
  "url": "/apps/cookbook/",
  "message": "explode(): Argument #2 ($string) must be of type string, array given",
  "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:94.0) Gecko/20100101 Firefox/94.0",
  "version": "22.2.3.0",
  "exception": {
    "Exception": "Exception",
    "Message": "explode(): Argument #2 ($string) must be of type string, array given",
    "Code": 0,
    "Trace": [
      {
        "file": "/config/www/nextcloud/lib/private/AppFramework/App.php",
        "line": 156,
        "function": "dispatch",
        "class": "OC\\AppFramework\\Http\\Dispatcher",
        "type": "->"
      },
      {
        "file": "/config/www/nextcloud/lib/private/Route/Router.php",
        "line": 302,
        "function": "main",
        "class": "OC\\AppFramework\\App",
        "type": "::"
      },
      {
        "file": "/config/www/nextcloud/lib/base.php",
        "line": 1006,
        "function": "match",
        "class": "OC\\Route\\Router",
        "type": "->"
      },
      {
        "file": "/config/www/nextcloud/index.php",
        "line": 36,
        "function": "handleRequest",
        "class": "OC",
        "type": "::"
      }
    ],
    "File": "/config/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
    "Line": 158,
    "Previous": {
      "Exception": "TypeError",
      "Message": "explode(): Argument #2 ($string) must be of type string, array given",
      "Code": 0,
      "Trace": [
        {
          "file": "/config/www/nextcloud/apps/cookbook/lib/Service/DbCacheService.php",
          "line": 300,
          "function": "explode"
        },
        {
          "file": "/config/www/nextcloud/apps/cookbook/lib/Service/DbCacheService.php",
          "line": 87,
          "function": "updateKeywords",
          "class": "OCA\\Cookbook\\Service\\DbCacheService",
          "type": "->",
          "args": [
            "*** sensitive parameters replaced ***"
          ]
        },
        {
          "file": "/config/www/nextcloud/apps/cookbook/lib/Service/DbCacheService.php",
          "line": 50,
          "function": "carryOutUpdate",
          "class": "OCA\\Cookbook\\Service\\DbCacheService",
          "type": "->"
        },
        {
          "file": "/config/www/nextcloud/apps/cookbook/lib/Service/DbCacheService.php",
          "line": 369,
          "function": "updateCache",
          "class": "OCA\\Cookbook\\Service\\DbCacheService",
          "type": "->",
          "args": [
            "*** sensitive parameters replaced ***"
          ]
        },
        {
          "file": "/config/www/nextcloud/apps/cookbook/lib/Service/DbCacheService.php",
          "line": 358,
          "function": "checkSearchIndexUpdate",
          "class": "OCA\\Cookbook\\Service\\DbCacheService",
          "type": "->"
        },
        {
          "file": "/config/www/nextcloud/apps/cookbook/lib/Controller/MainController.php",
          "line": 63,
          "function": "triggerCheck",
          "class": "OCA\\Cookbook\\Service\\DbCacheService",
          "type": "->"
        },
        {
          "file": "/config/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
          "line": 217,
          "function": "index",
          "class": "OCA\\Cookbook\\Controller\\MainController",
          "type": "->"
        },
        {
          "file": "/config/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
          "line": 126,
          "function": "executeController",
          "class": "OC\\AppFramework\\Http\\Dispatcher",
          "type": "->"
        },
        {
          "file": "/config/www/nextcloud/lib/private/AppFramework/App.php",
          "line": 156,
          "function": "dispatch",
          "class": "OC\\AppFramework\\Http\\Dispatcher",
          "type": "->"
        },
        {
          "file": "/config/www/nextcloud/lib/private/Route/Router.php",
          "line": 302,
          "function": "main",
          "class": "OC\\AppFramework\\App",
          "type": "::"
        },
        {
          "file": "/config/www/nextcloud/lib/base.php",
          "line": 1006,
          "function": "match",
          "class": "OC\\Route\\Router",
          "type": "->"
        },
        {
          "file": "/config/www/nextcloud/index.php",
          "line": 36,
          "function": "handleRequest",
          "class": "OC",
          "type": "::"
        }
      ],
      "File": "/config/www/nextcloud/apps/cookbook/lib/Service/DbCacheService.php",
      "Line": 300
    },
    "CustomMessage": "--"
  }
}
```

</details>

**Browser**
Happens on Firefox and Microsoft Edge

**Versions**
Nextcloud server version: 22.2.3 (Linuxserver docker container)
Cookbook version: 0.9.6
Database system: MariaDB

**Comment by christianlupus** [comment]
I see the most probable cause of the issue. However, I do not understand, why this is happening just now out of the blue. I suspect the NC server container was not updated recently, am I right @mhorst00? Are you using any 3rd party applications on the recipes (be it an android app or any software using the synchronized recipe folder)?

Could you as well search (on of) the recipe(s) that causes the issue? You should be able to find them quickly using the Linux command line:
```
grep '"keywords": *[[]' /path/to/nextcloud/folder/data/<user-name>/files/<recipe folder>/*/recipe.json
```
I suspect there will be at least one hit. I will prepare a PR to fix this. But it would be good to be sure this is the culprit.

**Comment by mhorst00** [comment]
Thank you for the answer. The container has been recently updated, but only to see if the error remains. I had the same thing on NC 22.1 and with PHP7. I then switched to PHP8 and then to a newer NC version with the same error. 
Yes, I am using a 3rd party Android app ([this one](https://f-droid.org/de/packages/de.micmun.android.nextcloudcookbook/)). 
With your `grep` I am seeing 3 recipes with hits, all of them have been added by the scraper inside the Android app and then synced to my server. Should I report this to the app developer as well?
